File: languages/java/src/main/java/com/bloock/sdk/entity/EciesKeyPair.java
package com.bloock.sdk.entity;

import com.bloock.sdk.bridge.proto.RecordOuterClass.GenerateEciesKeyPairResponse;

public class EciesKeyPair extends KeyPair {
  EciesKeyPair(String publicKey, String privateKey) {
    super(publicKey, privateKey);
  }

  public static EciesKeyPair fromProto(GenerateEciesKeyPairResponse keys) {
    return new EciesKeyPair(keys.getPublicKey(), keys.getPrivateKey());
  }
}

File: languages/php/Bloock/Entity/EciesKeyPair.php
<?php

namespace Bloock\Entity;

use Bloock\GenerateEcdsaKeysResponse;
use Bloock\GenerateEciesKeyPairResponse;

class EciesKeyPair extends KeyPair
{
    public function __construct(string $publicKey, string $privateKey)
    {
        parent::__construct($publicKey, $privateKey);
    }

    public static function fromProto(GenerateEciesKeyPairResponse $res): EciesKeyPair {
        return new EciesKeyPair($res->getPublicKey(), $res->getPrivateKey());
    }
}